app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,65 @@
+class TasksController < ApplicationController
+  def index
+    @all_tasks = Task.order(completed_at: :asc)#.where(completed_at: '')
+end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(tasks_create_params[:task])
+    if @task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    @task = Task.find(params[:id])
+    @task.update(tasks_create_params[:task])
+    redirect_to root_path
+  end
```

Comment:
Don't forget to handle the same error case from create here! Just in case the user is trying to update and the task doesn't save, you can redirect back to :edit